File: C/caesar-cipher.c


#include <stdio.h>
#include <stdlib.h>

struct cipherJob {
	int key, decision;
	char message[100];
};

typedef struct cipherJob CipherJob;
CipherJob getInput() {
	CipherJob userInput;

	printf("Caesar Cipher v1.0\n");
	printf("What do you want to do?\n");
	printf("\t [1] - Encode\n");
	printf("\t [2] - Decode\n");

	printf("Decision: ");
	scanf("%d", &userInput.decision);
	printf("\nEnter key: ");
	scanf("%d", &userInput.key);

	printf("%S", userInput.decision ? "Decryption" : "Encryption");

	

	return userInput;
}


int main() {
	CipherJob userInput = getInput();

	printf("decision: %d\n", userInput.decision);
	printf("key: %d\n", userInput.key);
	
	return 0;
}

File: Julia/caesar_cipher.jl

println("Caesar Cipher \n")
println("Commands: ")
println("\t - [1] Encrypt ")
println("\t - [2] Decrypt ")

println("\nWhat do you want to do?") 
print("Command: ")
decision = parse(Int8, readline())

alphabet = split("abcdefghijklmnopqrstuvwxyz", "")

if decision != 1 && decision != 2
    println("Invalid Command!")

elseif decision == 1
    println("\nEnter the text you want to encrypt")
    print("Text: ")
    text = readline()
    text = split(text, "")
    encrypted_array = Int8[]
    encrypted_text = Any[]

    print("Key: ")
    key = parse(Int8,readline())

    for i in 1:length(text)

        index = findfirst(x-> x == text[i], alphabet)

        if text[i] == " "
            append!(encrypted_text, " ")
        else
            append!(encrypted_array, (index + key) % 26)
        end
    
    end

    for i in 1:length(encrypted_array)
        if encrypted_array[i] == 0 
            index = 26
        else 
            index = i 
        end

        append!(encrypted_text, alphabet[index])
    end
    
    println("\nEncrypted Text: $(encrypted_text)")
    println(join(encrypted_text))
    println("Key $key")
    
end

    # for i in 1:length(text)
    #     text_array.append(char[])

